### PR TITLE
Brain MRI Super-Resolution Network

### PR DESCRIPTION
**Pull request by mhjos**
# Brain MRI Super-Resolution Network

## Description
This is a solution to Recognition task, and includes files related to project 5. 

Using the ADNI Brain MRI dataset, this super-resolution network is trained to generate up-scale images from 4x down-sampled input. The algorithm is based on an Efficient Sub-Pixel CNN.

## Files added
- `README.md` contains detailed information about the project, and example of usage.
- `dataset.py` contains the data loading and preprocessing for the Super-Resolution project.
- `modules.py` contains the Super-Resolution model implementation.
- `train.py` trains the Super-Resolution model using the ADNI brain dataset and saves the trained model.
- `predict.py` performs super-resolution on test data using the trained model and visualizes the results.

#
<Strong>Author:</Strong> MaiaJosang
<Strong>Student ID:</Strong> 46804158

### CODE REVIEW

**Comment by uqmarlonbran** [comment]
**This is an initial inspection - no action is required yet**

**Difficulty:** Normal

**General Comments:** Overall looks like a decent attempt. Most of the code is fine, just that separate datasets are produced for low and high rez images. They are then shuffled. This means that during training, two randomly paired images will be used to compute the MSE. README is a little light on details.

**[Recognition Problem]**
"modules.py":

- Seems to be a correct implementation.
- Very similar to Keras example, though does use relu.

"dataset.py":

- Resizes images to 240x240, just pads the image.
- Good use of ImageFolder to get images.
- Unfortunately, creates separate datasets for downsampled images and ground truth, and then shuffles them. As such, the loss is computed on two randomly paired images each iteration.

"train.py":

- Doesn't make use of validation set.
- Overall correct approach, but again uses two randomly paired images each iteration.
- Correctly tracks loss and plots at the end of training.
- Simply saves the final network at the end of training.

"predict.py":

- Doesn't compute metrics of the test dataset.
- Correctly loads model and provides some figures.

"README.MD":

- Provides a brief description of the network, though no figures to aid.
- Contains dependencies.
- Provides an example of inputs and outputs. Very low quality due to the problem during training.
- Provides a training plot, though again no validation or test losses.

**[Commit Log]**
Commit log seems ok.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem, data mixed within training leading to low performance. -2
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, no validation set -1
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, remove python compiled pyc files -2
Request Description good

**Comment by shakes76** [comment]
Please remove pyc files to recieve feedback marks +2

**Comment by mhjos** [comment]
Thank you, I removed the pyc files, as per the feedback.

**Comment by wangzhaomxy** [comment]
Feedback attempt and feedback marks granted +2.
